.github/workflows/continuous-integration.yml

```diff
@@ -1,17 +1,95 @@
 name: Continuous Integration
-on: [pull_request]
+on: [ pull_request ]
 
 jobs:
-  build:
-    name: Run unit tests with Jest
+  test:
+    name: Test
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout cplace-asc
+        uses: actions/checkout@v3
+
       - uses: actions/setup-node@v1
         with:
-          node-version: "12.x"
+          node-version: 12.x
+
+      - name: Cache Node Modules
+        id: npm-cache
+        uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/npm-shrinkwrap.json') }}
+
       - name: Install modules
-        run: npm install
+        run: npm ci
+
       - name: Run Jest
         run: node_modules/.bin/jest --verbose
 
+  compile:
+    env:
+      ENV_CPLACE_ARTIFACTORY_ACTOR: ${{ secrets.ENV_CPLACE_ARTIFACTORY_ACTOR }}
+      ENV_CPLACE_ARTIFACTORY_TOKEN: ${{ secrets.ENV_CPLACE_ARTIFACTORY_TOKEN }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        node_version:
+          - 12.x
+    name: Node ${{ matrix.node_version }} - ${{ matrix.architecture }} on ${{ matrix.os }}
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout cplace-asc
+        uses: actions/checkout@v3
+
+      - name: Checkout cplace
```

Comment:
should we also add a test for a product repo e.g. project-planning?

Reply:
Good idea! We should. @stefancplace could you add that?

Reply:
will do 👍